pkg/secret/replicated.go

```diff
@@ -46,25 +44,10 @@ func generateRandomString(charset string, length int) ([]byte, error) {
 }
 
 // DefaultSuperuserSecretName returns a default name for the superuser secret. It is of the general form
-// <cluster_name>-superuser.
+// <cluster_name>-superuser. Note that we expect clusterName to be a valid DNS subdomain name as defined in RFC 1123,
+// otherwise the secret name won't be valid.
 func DefaultSuperuserSecretName(clusterName string) string {
-	return fmt.Sprintf("%s-superuser", SanitizeSecretName(clusterName))
-}
-
-// SanitizeSecretName returns a string suitable for creating secret names derived from the given cluster name.
-// Secret names must be a valid DNS subdomain name as defined in RFC 1123.
-func SanitizeSecretName(name string) string {
```

Comment:
I don't think we need this anymore. The cluster name being the CRD name it has already been validated. If you try to create a k8c with an invalid name, you'd get:

```
The K8ssandraCluster "cluster1_fancy_name" is invalid: metadata.name: Invalid value: "cluster1_fancy_name": a lowercase RFC 1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')
```